core/src/main/java/org/kiwix/kiwixmobile/core/main/AnimatedProgressBar.kt

```diff
@@ -0,0 +1,238 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.os.Bundle
+import android.os.Parcelable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.animation.Animation
+import android.view.animation.DecelerateInterpolator
+import android.view.animation.Transformation
+import android.widget.LinearLayout
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.styleable
+
+class AnimatedProgressBar : LinearLayout {
+  private val mPaint = Paint()
+  private val mRect = Rect()
+  private var mProgress = 0
+  private var mBidirectionalAnimate = true
+  private var mDrawWidth = 0
+  private var mProgressColor = 0
+
+  constructor(context: Context, attrs: AttributeSet) : super(
+    context,
+    attrs
+  ) {
+    init(context, attrs)
+  }
+
+  constructor(
+    context: Context,
+    attrs: AttributeSet,
+    defStyleAttr: Int
+  ) : super(context, attrs, defStyleAttr) {
+    init(context, attrs)
+  }
+
+  /**
+   * Initialize the AnimatedProgressBar
+   *
+   * @param context is the context passed by the constructor
+   * @param attrs is the attribute set passed by the constructor
+   */
+  @Suppress("MagicNumber")
+  private fun init(
+    context: Context,
+    attrs: AttributeSet
+  ) {
+    val array = context.theme
+      .obtainStyledAttributes(attrs, styleable.AnimatedProgressBar, 0, 0)
+    val backgroundColor: Int
+    try {
+      // Retrieve the style of the progress bar that the user hopefully set
+      val defaultBackgroundColor = 0x424242
+      val defaultProgressColor = 0x2196f3
+      backgroundColor = array.getColor(
+        styleable.AnimatedProgressBar_backgroundColor, defaultBackgroundColor
+      )
+      mProgressColor = array.getColor(
+        styleable.AnimatedProgressBar_progressColor,
+        defaultProgressColor
+      )
+      mBidirectionalAnimate =
+        array.getBoolean(styleable.AnimatedProgressBar_bidirectionalAnimate, false)
+    } finally {
+      array.recycle()
+    }
+    val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+    inflater.inflate(R.layout.progress_bar, this, true)
```

Comment:
We have an extension method for inflating, you can just say `inflate(R.layout.progress_bar)`